bson-adt-mongo3/src/main/scala/adt/bson/mongo/client/BsonAdtCollection.scala

```diff
@@ -0,0 +1,130 @@
+package adt.bson.mongo.client
+
+import adt.bson._
+import com.mongodb.bulk.BulkWriteResult
+import com.mongodb.client._
+import com.mongodb.client.model._
+import com.mongodb.client.result.{DeleteResult, UpdateResult}
+import com.mongodb.{MongoNamespace, ReadPreference, WriteConcern}
+import org.bson.codecs.configuration.CodecRegistry
+
+import scala.collection.JavaConversions.{iterableAsScalaIterable, seqAsJavaList}
+import scala.language.implicitConversions
+
+object BsonAdtCollection {
+
+  def apply(collection: MongoCollection[BsonObject]): BsonAdtCollection = new BsonAdtCollection(collection)
```

Comment:
Do you mean, why wrap the collection and not extend `MongoCollection`? Or why have an apply method instead of just the constructor?